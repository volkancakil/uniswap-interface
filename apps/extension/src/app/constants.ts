import { SpaceTokens } from 'ui/src'

export const SCREEN_ITEM_HORIZONTAL_PAD = '$spacing12' satisfies SpaceTokens
