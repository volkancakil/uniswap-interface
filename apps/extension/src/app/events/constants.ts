export enum GlobalErrorEvent {
  ReduxStorageExceeded = 'ReduxStorageExceeded',
}
