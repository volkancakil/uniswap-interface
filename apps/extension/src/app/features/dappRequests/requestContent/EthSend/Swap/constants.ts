import { BigNumber } from 'ethers'

export const CONTRACT_BALANCE = BigNumber.from(2).pow(255)
export const ETH_ADDRESS = '0x0000000000000000000000000000000000000000'
export const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
export const MAX_UINT256 = BigNumber.from(2).pow(256).sub(1)
export const MAX_UINT160 = BigNumber.from(2).pow(160).sub(1)

export const SENDER_AS_RECIPIENT = '0x0000000000000000000000000000000000000001'
export const ROUTER_AS_RECIPIENT = '0x0000000000000000000000000000000000000002'

export const OPENSEA_CONDUIT_SPENDER_ID = 0
export const SUDOSWAP_SPENDER_ID = 1
