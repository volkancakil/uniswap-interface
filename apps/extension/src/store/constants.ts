export const PERSIST_KEY = 'root'
export const STATE_STORAGE_KEY = `persist:${PERSIST_KEY}`
