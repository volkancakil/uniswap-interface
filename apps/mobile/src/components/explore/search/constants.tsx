import { SearchHeader, SearchHeaderKey } from 'src/components/explore/search/types'
import { Coin, Gallery, Person } from 'ui/src/components/icons'
import { iconSizes } from 'ui/src/theme'
import { getChainInfo } from 'uniswap/src/features/chains/chainInfo'
import { UniverseChainId } from 'uniswap/src/features/chains/types'
import i18n from 'uniswap/src/i18n'

export const SEARCH_RESULT_HEADER_KEY: SearchHeaderKey = 'header'

export const SEARCH_ITEM_PX = '$spacing20'
export const SEARCH_ITEM_PY = '$spacing8'
export const SEARCH_ITEM_ICON_SIZE = iconSizes.icon36

const ICON_SIZE = '$icon.24'
const ICON_COLOR = '$neutral2'

export const WalletHeaderItem: SearchHeader = {
  icon: <Person color={ICON_COLOR} size={ICON_SIZE} />,
  type: SEARCH_RESULT_HEADER_KEY,
  title: i18n.t('explore.search.section.wallets'),
}
export const TokenHeaderItem: SearchHeader = {
  icon: <Coin color={ICON_COLOR} size={ICON_SIZE} />,
  type: SEARCH_RESULT_HEADER_KEY,
  title: i18n.t('explore.search.section.tokens'),
}
export const NFTHeaderItem: SearchHeader = {
  icon: <Gallery color={ICON_COLOR} size={ICON_SIZE} />,
  type: SEARCH_RESULT_HEADER_KEY,
  title: i18n.t('explore.search.section.nft'),
}
export const EtherscanHeaderItem: (chainId: UniverseChainId) => SearchHeader = (chainId: UniverseChainId) => ({
  type: SEARCH_RESULT_HEADER_KEY,
  title: i18n.t('explore.search.action.viewEtherscan', {
    blockExplorerName: getChainInfo(chainId).explorer.name,
  }),
})
