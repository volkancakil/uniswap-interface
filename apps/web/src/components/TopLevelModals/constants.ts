export const LAUNCH_MODAL_DESKTOP_MAX_WIDTH = 520
export const LAUNCH_MODAL_DESKTOP_MAX_HEIGHT = 320
export const LAUNCH_MODAL_MOBILE_MAX_HEIGHT = 564
export const LAUNCH_MODAL_MOBILE_MAX_IMAGE_HEIGHT = 392
