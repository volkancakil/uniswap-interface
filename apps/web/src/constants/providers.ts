import AppJsonRpcProvider from 'rpc/AppJsonRpcProvider'
import ConfiguredJsonRpcProvider from 'rpc/ConfiguredJsonRpcProvider'
import { getChainInfo } from 'uniswap/src/features/chains/chainInfo'
import { SUPPORTED_CHAIN_IDS, UniverseChainId } from 'uniswap/src/features/chains/types'

function getAppProvider(chainId: UniverseChainId) {
  const info = getChainInfo(chainId)
  return new AppJsonRpcProvider(
    info.rpcUrls.interface.http.map((url) => new ConfiguredJsonRpcProvider(url, { chainId, name: info.interfaceName })),
  )
}

/** These are the only JsonRpcProviders used directly by the interface. */
export const RPC_PROVIDERS = Object.fromEntries(
  SUPPORTED_CHAIN_IDS.map((chain) => [chain as UniverseChainId, getAppProvider(chain)]),
) as Record<UniverseChainId, AppJsonRpcProvider>
