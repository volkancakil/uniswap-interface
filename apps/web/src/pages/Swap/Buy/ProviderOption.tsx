import { useMemo } from 'react'
import { useAddFiatOnRampTransaction } from 'state/fiatOnRampTransactions/hooks'
import { FiatOnRampTransactionStatus, FiatOnRampTransactionType } from 'state/fiatOnRampTransactions/types'
import { ExternalLink } from 'ui/src/components/icons/ExternalLink'
import { UNISWAP_WEB_URL } from 'uniswap/src/constants/urls'
import { FORQuoteItem } from 'uniswap/src/features/fiatOnRamp/FORQuoteItem'
import { useFiatOnRampAggregatorWidgetQuery } from 'uniswap/src/features/fiatOnRamp/api'
import { FORCountry, FORQuote, FORServiceProvider, FiatCurrencyInfo } from 'uniswap/src/features/fiatOnRamp/types'
import { createOnRampTransactionId } from 'uniswap/src/features/fiatOnRamp/utils'
import { FiatOnRampEventName } from 'uniswap/src/features/telemetry/constants'
import { sendAnalyticsEvent } from 'uniswap/src/features/telemetry/send'

interface ProviderOptionProps {
  quote: FORQuote
  selectedCountry: FORCountry
  quoteCurrencyCode: string
  inputAmount: string
  meldSupportedFiatCurrency: FiatCurrencyInfo
  walletAddress: string
  setConnectedProvider: (provider: FORServiceProvider) => void
  setErrorProvider: (provider: FORServiceProvider) => void
}

export function ProviderOption({
  quote,
  selectedCountry,
  quoteCurrencyCode,
  inputAmount,
  meldSupportedFiatCurrency,
  walletAddress,
  setConnectedProvider,
  setErrorProvider,
}: ProviderOptionProps) {
  const addFiatOnRampTransaction = useAddFiatOnRampTransaction()

  const widgetQueryParams = useMemo(() => {
    return {
      serviceProvider: quote.serviceProviderDetails.serviceProvider,
      countryCode: selectedCountry.countryCode,
      destinationCurrencyCode: quoteCurrencyCode,
      sourceAmount: parseFloat(inputAmount),
      sourceCurrencyCode: meldSupportedFiatCurrency.code,
      walletAddress,
      externalSessionId: createOnRampTransactionId(quote.serviceProviderDetails.serviceProvider),
      redirectUrl: `${UNISWAP_WEB_URL}/buy`,
    }
  }, [
    inputAmount,
    meldSupportedFiatCurrency.code,
    quote.serviceProviderDetails.serviceProvider,
    quoteCurrencyCode,
    selectedCountry.countryCode,
    walletAddress,
  ])

  // TODO(WEB-4417): use the widgetUrl from the /quote response instead of prefetching for every provider.
  const { data, error, isLoading } = useFiatOnRampAggregatorWidgetQuery(widgetQueryParams)

  return (
    <FORQuoteItem
      key={quote.serviceProviderDetails.serviceProvider}
      serviceProvider={quote.serviceProviderDetails}
      hoverIcon={<ExternalLink position="absolute" right="$spacing12" size={20} />}
      isLoading={isLoading}
      onPress={async () => {
        if (data) {
          window.open(data.widgetUrl, '_blank')
          setConnectedProvider(quote.serviceProviderDetails)
          addFiatOnRampTransaction({
            externalSessionId: widgetQueryParams.externalSessionId,
            account: walletAddress,
            status: FiatOnRampTransactionStatus.INITIATED,
            forceFetched: false,
            addedAt: Date.now(),
            type: FiatOnRampTransactionType.BUY,
            syncedWithBackend: false,
            provider: quote.serviceProviderDetails.serviceProvider,
          })
          sendAnalyticsEvent(FiatOnRampEventName.FiatOnRampWidgetOpened, {
            countryCode: selectedCountry.countryCode,
            countryState: selectedCountry.state,
            cryptoCurrency: quoteCurrencyCode,
            externalTransactionId: widgetQueryParams.externalSessionId,
            fiatCurrency: meldSupportedFiatCurrency.code,
            serviceProvider: quote.serviceProviderDetails.serviceProvider,
          })
        } else if (error) {
          setErrorProvider(quote.serviceProviderDetails)
        }
      }}
    />
  )
}
