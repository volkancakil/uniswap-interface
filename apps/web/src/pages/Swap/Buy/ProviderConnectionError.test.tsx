import { ProviderConnectionError } from 'pages/Swap/Buy/ProviderConnectionError'
import { mockServiceProvider } from 'pages/Swap/Buy/test/constants'
import { fireEvent, render, screen } from 'test-utils/render'

describe('ProviderConnectionError', () => {
  it('should render the component and call callbacks', () => {
    const onBack = jest.fn()
    const closeModal = jest.fn()

    const { container } = render(
      <ProviderConnectionError onBack={onBack} closeModal={closeModal} selectedServiceProvider={mockServiceProvider} />,
    )

    expect(container.firstChild).toMatchSnapshot()

    fireEvent.click(screen.getByTestId('ConnectingViewWrapper-close'))
    expect(closeModal).toHaveBeenCalled()

    fireEvent.click(screen.getByTestId('ConnectingViewWrapper-back'))
    expect(onBack).toHaveBeenCalled()

    screen.getByText('Something went wrong connecting with Test Provider.')
  })
})
