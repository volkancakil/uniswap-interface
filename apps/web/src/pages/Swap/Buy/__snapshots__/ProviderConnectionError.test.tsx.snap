// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderConnectionError should render the component and call callbacks 1`] = `
<span
  class=""
  style="display: contents;"
>
  <span
    class=" _dsp_contents"
  >
    <span
      class=" t_light _dsp_contents is_Theme"
    >
      <div
        class="_pr-_sm_t-space-spa94665593 _pl-_sm_t-space-spa94665593 _pb-_sm_t-space-spa1360334076 _display-flex _alignItems-stretch _flexBasis-auto _boxSizing-border-box _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column _gap-t-space-spa1360334076 _position-relative"
      >
        <svg
          style="position: absolute; z-index: 0; width: 100%; height: 100%; color: rgb(125, 125, 125);"
        >
          for-connecting-v2.svg
        </svg>
        <div
          class="_display-flex _alignItems-stretch _flexBasis-auto _boxSizing-border-box _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column _position-absolute _zIndex-1 _width-10037 _height-10037 _top-0px _left-0px _background-radial-grad1116705115"
        />
        <div
          class="_display-flex _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-row-reverse _alignItems-center _justifyContent-space-betwe3241 _zIndex-2"
        >
          <div
            class="_opacity-0active-0--75 _transform-0active-scale11281 _display-flex _alignItems-stretch _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column _cursor-pointer"
            data-testid="ConnectingViewWrapper-close"
          >
            <div
              class="_display-flex _alignItems-stretch _flexDirection-column _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0"
            >
              <svg
                fill="none"
                stroke-width="8"
                style="width: 24px; height: 24px; color: rgb(125, 125, 125);"
                viewBox="0 0 16 16"
              >
                <path
                  d="M12.5303 4.53033C12.8232 4.23744 12.8232 3.76256 12.5303 3.46967C12.2374 3.17678 11.7626 3.17678 11.4697 3.46967L12.5303 4.53033ZM3.46967 11.4697C3.17678 11.7626 3.17678 12.2374 3.46967 12.5303C3.76256 12.8232 4.23744 12.8232 4.53033 12.5303L3.46967 11.4697ZM4.53033 3.46967C4.23744 3.17678 3.76256 3.17678 3.46967 3.46967C3.17678 3.76256 3.17678 4.23744 3.46967 4.53033L4.53033 3.46967ZM11.4697 12.5303C11.7626 12.8232 12.2374 12.8232 12.5303 12.5303C12.8232 12.2374 12.8232 11.7626 12.5303 11.4697L11.4697 12.5303ZM11.4697 3.46967L3.46967 11.4697L4.53033 12.5303L12.5303 4.53033L11.4697 3.46967ZM3.46967 4.53033L11.4697 12.5303L12.5303 11.4697L4.53033 3.46967L3.46967 4.53033Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
          <svg
            cursor="pointer"
            data-testid="ConnectingViewWrapper-back"
            fill="#7D7D7D"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="19"
              x2="5"
              y1="12"
              y2="12"
            />
            <polyline
              points="12 19 5 12 12 5"
            />
          </svg>
        </div>
        <div
          class="_display-flex _alignItems-stretch _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column _mt-t-space-spa1360333989 _zIndex-2 _width-10037 _height-10037"
        >
          <div
            class="_display-flex _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column _alignItems-center _gap-t-space-spa1360334014"
          >
            <div
              class="_display-flex _alignItems-stretch _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-row _gap-t-space-spa1360334076"
            >
              <div
                class="_display-flex _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column _alignItems-center _justifyContent-center"
                style="background-color: rgb(255, 239, 248); border-radius: 20px; height: 90px; width: 90px;"
              >
                <div
                  class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
                  style="height: 64px; width: 64px;"
                >
                  <div
                    class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
                    style="background-image: url(uniswap-logo-large.png);"
                  />
                  <img
                    alt=""
                    class="css-accessibilityImage-9pa8cd"
                    draggable="false"
                    src="uniswap-logo-large.png"
                  />
                </div>
              </div>
              <img
                height="120"
                src="test-provider-logo-light"
                style="background-color: rgb(255, 239, 248); border-radius: 20px; height: 90px; width: 90px;"
                width="120"
              />
            </div>
            <div
              class="_display-flex _alignItems-center _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column _justifyContent-center _gap-t-space-spa94665593"
            >
              <span
                class="font_subHeading _display-inline _boxSizing-border-box _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _color-statusCriti98254 _fontFamily-f-family _wordWrap-break-word _fontSize-f-size-larg101 _lineHeight-f-lineHeigh1262266530 _fontWeight-f-weight-bo3548"
                data-disable-theme="true"
              >
                Connection failed
              </span>
              <span
                class="font_body _display-inline _boxSizing-border-box _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _color-neutral2 _fontFamily-f-family _wordWrap-break-word _fontSize-f-size-medi3736 _lineHeight-f-lineHeigh507465454 _fontWeight-f-weight-bo3548 _textAlign-center"
                data-disable-theme="true"
              >
                Something went wrong connecting with Test Provider.
              </span>
            </div>
            <span
              class=" _dsp_contents"
            >
              <span
                class="t_sub_theme t_primary_Button _dsp_contents is_Theme"
              >
                <button
                  class="is_Button _backgroundColor-0hover-accent3Hove112785 _backgroundColor-0active-backgroundP3496915 _opacity-0active-0--7 _display-flex _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _borderTopWidth-1px _borderRightWidth-1px _borderBottomWidth-1px _borderLeftWidth-1px _flexDirection-row _alignItems-center _justifyContent-center _borderTopColor-borderColor _borderRightColor-borderColor _borderBottomColor-borderColor _borderLeftColor-borderColor _cursor-pointer _height-auto _pt-t-space-spa1360334080 _pr-t-space-spa1360334080 _pb-t-space-spa1360334080 _pl-t-space-spa1360334080 _borderTopLeftRadius-t-radius-ro291586449 _borderTopRightRadius-t-radius-ro291586449 _borderBottomRightRadius-t-radius-ro291586449 _borderBottomLeftRadius-t-radius-ro291586449 _gap-t-space-spa94665593 _backgroundColor-accent3 _width-10037 _borderBottomStyle-solid _borderTopStyle-solid _borderLeftStyle-solid _borderRightStyle-solid"
                  data-disable-theme="true"
                >
                  Try again
                </button>
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        style="display: contents;"
      />
    </span>
  </span>
</span>
`;
